src/program/protocols/union/user_take.rs

```diff
@@ -253,6 +255,20 @@ impl UserTakeProtocol {
             .as_ref()
             .to_vec();
 
+        if user_take_sighash != pegout_request.pegout_signature_hash {
+            let error = ProtocolError {
+                uuid: self.ctx.id,
+                protocol: ProtocolName::UserTake,
+                source: ProtocolErrorType::InvalidSighash {
+                    expected: pegout_request.pegout_signature_hash,
+                    found: user_take_sighash.clone(),
+                },
+            };
+            let data = send_to_l2!(self, program_context, ProtocolError, error);
+            info!(id = self.ctx.my_idx, "Failed to accept user take: {}", data);
+            return Err(BitVMXError::InvalidParameter(error.to_string()));
+        }
+
```

Comment:
Now that we have the check, I suggest moving it at the beginning of this function for fast failing (along with accessing `user_take_sighash` from BitVMX for the check)